README.md

```diff
@@ -1 +1,167 @@
+[david-dm-url]:https://david-dm.org/moxystudio/webpack-sane-compiler
+[david-dm-image]:https://img.shields.io/david/moxystudio/webpack-sane-compiler.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/webpack-sane-compiler?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/webpack-sane-compiler.svg
+[greenkeeper-image]:https://badges.greenkeeper.io/moxystudio/webpack-sane-compiler.svg
+[greenkeeper-url]:https://greenkeeper.io
+
+A webpack compiler wrapper that provides a nicer API.
+
+
+## Installation
+
+`$ npm install webpack-sane-compiler --save-dev`
+
+## Usage
+
+```js
+const webpack = require('webpack');
+const saneWebpackCompiler = require('webpack-sane-compiler');
+
+const webpackCompiler = webpack(/* config */);
+const compiler = saneWebpackCompiler(webpackCompiler);
+```
+
+Alternatively, you may pass a config directly instead of a webpack compiler:
+
+```js
+const compiler = saneWebpackCompiler(/* config */);
+```
+
+The compiler inherits from [EventEmitter](https://nodejs.org/api/events.html) and emits the following events:
+
+| Name   | Description   | Arguments |
+| ------ | ------------- | -------- |
+| begin | Emitted when a compilation starts | |
+| error | Emitted when the compilation fails | (`err: Error`) |
+| end | Emitted when the compilation completes successfully | (`stats: WebpackStats`) |
+
+```js
+compiler
+.on('begin', () => console.log('Compilation started'))
+.on('end', (stats) => {
+    console.log('Compilation finished successfully');
+    console.log('Stats', stats);
+})
+.on('error', (err) => {
+    console.log('Compilation failed')
+    console.log(err.message);
+    console.log(err.stats.toString());
+})
+```
+
+### .run()
+
+Returns a Promise that fulfills with a `stats` object or is rejected with an error.
+
+This is similar to webpack's run() method, except that it returns a promise which gets rejected if stats contains errors.
+
+```js
+compiler.run()
+.then((stats) => {
+    // do something with stats
+})
+.catch((err) => {
+    // err = {
+    //   message: 'Error message',
+    //   [stats]: <webpack-stats>
+    // }
+});
+```
+
+### .watch([options], [handler])
+
+Starts watching for changes and compiles on-the-fly.   
+Returns itself to allow chaining.
+
+Calls `handler` everytime the compilation fails or succeeds.
+This is similar to webpack's watch() method, except that `handler` gets called with an error if stats contains errors.
+
+Available options:
+
+| Name   | Description   | Type     | Default |
+| ------ | ------------- | -------- | ------- |
+| poll | Use polling instead of native watchers | boolean | `false` |
+| aggregateTimeout | Wait so long for more changes (ms) | err | 200 |
+
+```js
+compiler.watch((err, stats) => {
+    // err = {
+    //   message: 'Error message',
+    //   [stats]: <webpack-stats>
+    // }
+});
+```
+
+### .unwatch()
+
+Stops watching for changes.   
+Returns a promise that fulfills when done.
+
+
+### .resolve()
```

Comment:
`.resolve()` is not implemented in the source

Reply:
Done